File: app/src/main/java/com/jeeps/gamecollector/comparators/GameByNameComparator.java
package com.jeeps.gamecollector.comparators;

import com.jeeps.gamecollector.model.Game;

import java.util.Comparator;

/**
 * Created by jeeps on 1/9/2018.
 */

public class GameByNameComparator implements Comparator<Game> {

    private boolean desc;

    public GameByNameComparator(boolean desc) {
        this.desc = desc;
    }

    public GameByNameComparator() {
        this.desc = false;
    }

    @Override
    public int compare(Game game1, Game game2) {
        if (!desc)
            return game1.getName().compareTo(game2.getName());
        return (game1.getName().compareTo(game2.getName())) * -1;
    }
}

File: app/src/main/java/com/jeeps/gamecollector/remaster/utils/comparators/GameByNameComparator.kt
package com.jeeps.gamecollector.remaster.utils.comparators

import com.jeeps.gamecollector.remaster.data.model.data.games.Game

/**
 * Created by jeeps on 1/9/2018.
 */
class GameByNameComparator : Comparator<Game> {
    private var desc: Boolean

    constructor(desc: Boolean) {
        this.desc = desc
    }

    constructor() {
        desc = false
    }

    override fun compare(game1: Game, game2: Game): Int {
        return if (!desc)
            game1.name.compareTo(game2.name, true)
        else
            game1.name.compareTo(game2.name, true) * -1
    }
}